tests/suites/test_suite_psa_crypto.data

```diff
@@ -1038,8 +1038,8 @@ depends_on:PSA_WANT_ALG_CTR:PSA_WANT_KEY_TYPE_AES:MBEDTLS_PSA_CRYPTO_STORAGE_C
 copy_fail:PSA_KEY_USAGE_COPY | PSA_KEY_USAGE_ENCRYPT | PSA_KEY_USAGE_EXPORT:PSA_ALG_CTR:0:PSA_KEY_TYPE_AES:"404142434445464748494a4b4c4d4e4f":PSA_KEY_TYPE_AES:0:PSA_KEY_USAGE_COPY | PSA_KEY_USAGE_ENCRYPT | PSA_KEY_USAGE_EXPORT:PSA_ALG_CTR:0:0:PSA_KEY_LIFETIME_PERSISTENT:PSA_ERROR_INVALID_ARGUMENT
 
 Copy fail: AES, invalid lifetime (unknown location) in attributes
-depends_on:PSA_WANT_ALG_CTR:PSA_WANT_KEY_TYPE_AES:MBEDTLS_PSA_CRYPTO_STORAGE_C
-copy_fail:PSA_KEY_USAGE_COPY | PSA_KEY_USAGE_ENCRYPT | PSA_KEY_USAGE_EXPORT:PSA_ALG_CTR:0:PSA_KEY_TYPE_AES:"404142434445464748494a4b4c4d4e4f":PSA_KEY_TYPE_AES:0:PSA_KEY_USAGE_COPY | PSA_KEY_USAGE_ENCRYPT | PSA_KEY_USAGE_EXPORT:PSA_ALG_CTR:0:1:PSA_KEY_LIFETIME_FROM_PERSISTENCE_AND_LOCATION(PSA_KEY_LIFETIME_PERSISTENT, 11):PSA_ERROR_INVALID_ARGUMENT
+depends_on:PSA_WANT_ALG_CTR:PSA_WANT_KEY_TYPE_AES:MBEDTLS_PSA_CRYPTO_STORAGE_C:!MBEDTLS_PSA_CRYPTO_DRIVERS
```

Comment:
If I understand https://github.com/ARMmbed/mbedtls/pull/4766/commits/0d48780ae8aafda7ac6117b7a8d0ddec22c0f05a#r683179710 correctly, with `MBEDTLS_PSA_CRYPTO_DRIVERS` defined, copying across locations isn't supported, so the attempt yields `PSA_ERROR_NOT_SUPPORTED` instead of `PSA_ERROR_INVALID_ARGUMENT`. And there's a test case for NOT_SUPPORTED in `test_suite_psa_crypto_driver_wrappers.data`. Ok, but please add a test case where the lifetime is invalid due to an invalid persistence level in the default location. This should still invoke the `INVALID_ARGUMENT` code path and so should be independent of opaque driver support.

Reply:
`Ok, but please add a test case where the lifetime is invalid due to an invalid persistence level in the default location`
There already is a test to cover invalid persistence for default location(Copy fail: AES, invalid persistent key identifier in attributes). Is this something else ?

Reply:
Invalid identifier and invalid persistence level are different error cases that don't necessarily go through the same code paths. The test case you removed had a lifetime which was invalid due to having an unsupported location. Your PR makes all locations potentially supported (if a driver is available at that location), which removes one reason for the lifetime to be invalid. But the other reason for an invalid lifetime (invalid persistence level) remains.

Reply:
As discussed over slack, `psa_validate_key_persistence` does not do much with specific levels of persistence. So additional testcases do nothing. Included a missing test case to validate failing while copying to a readonly lifetime.